kolibri/plugins/coach/test/test_coach_api.py

```diff
@@ -98,6 +99,49 @@ def assert_progress(node, progress):
             [{'log_count_complete': 0, 'log_count_total': 1, 'total_progress': 0.5}]
         ])
 
+    def test_contentreport_time_filtering(self):
+
+        # set up data for testing time filtering on content node endpoint
+        facility = Facility.objects.create(name="MyFac")
+        user = FacilityUser.objects.create(username="learner", facility=facility)
+        user.set_password("pass")
+        user.save()
+
+        admin = FacilityUser.objects.create(username="admin", facility=facility)
+        admin.set_password("pass")
+        admin.save()
+
+        Role.objects.create(user=admin, collection=facility, kind=role_kinds.ADMIN)
+        root = content.ContentNode.objects.get(title="root")
+        c2c1 = content.ContentNode.objects.get(title="c2c1")
+        c2c3 = content.ContentNode.objects.get(title="c2c3")
+
+        start_date = datetime.datetime(2000, 1, 1)
+
+        date_1 = timezone.now() - datetime.timedelta(8)
+
+        date_2 = timezone.now() - datetime.timedelta(6)
+
+        for date, node in ((date_1, c2c1), (date_2, c2c3)):
+            ContentSummaryLog.objects.create(
+                user=user,
+                content_id=node.content_id,
+                progress=1.0,
+                kind=node.kind,
+                channel_id=self.the_channel_id,
+                start_timestamp=start_date,
+                end_timestamp=date,
+            )
+
+        # check that only the log less than 7 days ago returns from recent report
+        self.client.login(username="admin", password="pass", facility=facility)
+        response = self.client.get(self._reverse_channel_url("kolibri:coach:recentreport-list", {
+            'content_node_id': root.id,
+            'collection_kind': collection_kinds.FACILITY,
+            'collection_id': facility.id,
+        }))
+        self.assertEqual(len(response.data), 1)
```

Comment:
Perhaps instead of checking the length, check the values returned. For example, checking that the correct `content_id` is returned in the response.